mlflow/models/model.py

```diff
@@ -191,7 +197,14 @@ def to_dict(self):
 
     def to_yaml(self, stream=None):
         """Write the model as yaml string."""
-        return yaml.safe_dump(self.to_dict(), stream=stream, default_flow_style=False)
+
+        # Do not save mlflow_version attribute if it's not defined
+        info = self.to_dict()
```

Comment:
Can we move the logic for discarding `mlflow_version` if it's `None` into `to_dict()`?

Reply:
Surely can do but my thinking was that a new version of model code (or some external code using it) "knows" object has a property named `mlflow_version` so logically may also expect the same property in the model info and call it without checking if it exists. Also, `to_dict` could be called outside of the `save` scope. Let me know if you think moving it to `to_dict` is still a good idea and I'll do it.